src/Microsoft.Diagnostics.Monitoring.StartupHook/DiagnosticsBootstrapper.cs

```diff
@@ -15,25 +15,28 @@ namespace Microsoft.Diagnostics.Monitoring.StartupHook
     internal sealed class DiagnosticsBootstrapper :
         IDisposable
     {
-        private readonly CurrentAppDomainExceptionProcessor _exceptionProcessor = new();
+        private readonly CurrentAppDomainExceptionProcessor _exceptionProcessor;
         private readonly AspNetHostingStartupLoader? _hostingStartupLoader;
 
         private long _disposedState;
 
         public DiagnosticsBootstrapper()
         {
+            _exceptionProcessor = new(ToolIdentifiers.IsEnvVarEnabled(InProcessFeaturesIdentifiers.EnvironmentVariables.Exceptions.IncludeMonitorExceptions));
+            _exceptionProcessor.Start();
```

Comment:
We won't be able to dynamically change the value of IncludeMonitorExceptions with this approach. There might be some ways to do that, but does not have to be a part of this PR.